### DISCUSSION TITLE
Consolidate multiple summary reports of Artillery

### DISCUSSION THREAD

**Discussion by priya-dsp** [artilleryio/artillery#2479] [Artillery]
When we run multiple yamls either sequentially or parallelly, seperate summary reports are getting generated. Instead of having multiple reports, we need one single report. How to combine or generate single report for multiple yamls?

If we generate combined summary report , we are getting values picked up from recent report for the below parameters:
http.codes.200
http.downloaded_bytes
http.requests
http.responses
plugins.expect.ok
plugins.expect.ok.contentType
plugins.expect.ok.equals
plugins.expect.ok.hasProperty
vusers.completed
vusers.created
vusers.failed

I'm not using Artillery pro.

<img width="580" alt="Combined report" src="https://github.com/artilleryio/artillery/assets/159108432/97add888-4b1b-4ac7-aed8-86efa76c6c04">

**Comment by hassy** [comment]
hi @priya-dsp 👋 What are you trying to accomplish by running multiple scripts? Artillery does not support merging multiple reports from different runs.

**Reply by priya-dsp** [reply]
hi @hassy , we have different module's/scripts of API's included in each yamls.  So we have written batch file to run each of those module's. As summary report is generated for each yamls separately when we run those scripts. Instead of having multiple summary reports for each module's. We are trying to consolidate all the summary reports of different module's into one single report. Is there any alternative for this?

I have tried writing a Javascript to parse those multiple json reports and generate combined report. However, its partially working. The data of default metrics on summary report listed above in screenshot are not getting merged.
